lardataobj/RawData/raw.cxx

```diff
@@ -1386,21 +1315,84 @@ namespace raw {
       // and if the last 2 number are one
       // and if we are not at the end
       if ((current_number.size()>=2 && it >= 1 && comp[it-1] == 1 && comp[it] == 1) || it == comp.size()-1) {
-        short zigzag_number = 0;
-        for (size_t it2=0; it2<current_number.size()-1; ++it2) {
-          if (current_number[it2])
-            zigzag_number += current_number[it2] * FibNumbers[it2+1];
-        }
+        current_number.pop_back();
+        short zigzag_number = decode_table_chunk(current_number);
+      
         short decoded = 0;
         if (zigzag_number%2 == 0) decoded = zigzag_number / 2;
         else                      decoded = -(zigzag_number - 1) / 2;
         short baseline = uncompressed.back();
+      
         uncompressed.push_back(baseline + decoded);
+      
         current_number.clear();
         if (uncompressed.size() == n_samples) break;
       }
     }
-  
+    return;
   }
 
+  void fibonacci_encode_table(int end, std::vector<std::vector<bool>>& table) {
+    if ((int)table.size() > end)
+      return;
+
+    std::map<int, int> fibn; // no need for this to be static
+    fibn[0] = 0;
+    fibn[1] = 1;
+    fibn[2] = 1;
+                   
+    // if the table was empty, fill it
+    if (table.size()==0) {
+      table.push_back(std::vector<bool>()); // we need this one so that the index of the vector is the same as the number we want to encode
+      table.push_back({true      });
+      table.push_back({false,true});
+    }
+
+    if ((int)table.size() > end)
+      return;
+
+    // start at the third fibonacci number
+    for (int i=2; fibn.rbegin()->second<end; ++i) {
+      fibn[i] = fibn[i-1] + fibn[i-2];
+    }
+                   
+
+    for (int i=table.size(); i<=end; ++i) {
+      int current_number = i;
+      std::vector<bool> seq; // the final sequence of numbers
+      while (current_number>0) {
+        // find the biggest fibonacci number that is smaller than the current number
+        for (auto fib = fibn.rbegin(); fib != fibn.rend(); ++fib) {
```

Comment:
Same comment as in original commit -- use std::lower_bound to make this a logarithmic-time loop instead of a linear time loop

Reply:
As Tom has already approved this PR, I will resolve this issue.

Reply:
Ah, sorry I left this out.

Reply:
No worries.  Up to you if you want to get it in there for this week.  You can also wait until this PR has been merged in, and open up another PR with the lower_bound optimization.